File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int n = nums.length;
        Set<Integer> seen = new HashSet<>();
        for(int num: nums) {
            seen.add(num);
        }

        int longest = 0;

        for(int num: nums) {
            if(!seen.contains(num-1)) {
                int length = 1;
                while(seen.contains(num + length)) {
                    length++;
                }
                longest = Math.max(longest, length);
            }
        }

        return longest;
    }
}

File: Data Structures & Algorithms/max-water-container/submission-2.py
class Solution:
    def maxArea(self, nums: List[int]) -> int:
        max_area = 0

        for i in range(len(nums)):
            for j in range(i+1, len(nums)):
                area = min(nums[i], nums[j]) * (j-i)

                max_area = max(max_area, area)
        
        return max_area